go-controller/pkg/ovn/egressip.go

```diff
@@ -416,47 +537,27 @@ func (e *egressIPMode) createEgressPolicy(podIps []net.IP, status egressipv1.Egr
 		} else if !isEgressIPv6 && !utilnet.IsIPv6(podIP) {
 			filterOption = fmt.Sprintf("ip4.src == %s", podIP.String())
 		}
-		policyIDs, err := findLRPolicyIDs(filterOption, egressIPName)
+		policyIDs, err := findReroutePolicyIDs(filterOption, egressIPName, gatewayRouterIP)
 		if err != nil {
 			return err
 		}
 		if policyIDs == nil {
-			if packetMark != 0 {
-				_, stderr, err = util.RunOVNNbctl(
-					"--id=@lr-policy",
-					"create",
-					"logical_router_policy",
-					"action=reroute",
-					fmt.Sprintf("match=\"%s\"", filterOption),
-					fmt.Sprintf("priority=%v", egressIPReroutePriority),
-					fmt.Sprintf("nexthop=%s", gatewayRouterIP),
-					fmt.Sprintf("external_ids:name=%s", egressIPName),
-					fmt.Sprintf("options:pkt_mark=%v", packetMark),
-					"--",
-					"add",
-					"logical_router",
-					ovnClusterRouter,
-					"policies",
-					"@lr-policy",
-				)
-			} else {
-				_, stderr, err = util.RunOVNNbctl(
-					"--id=@lr-policy",
-					"create",
-					"logical_router_policy",
-					"action=reroute",
-					fmt.Sprintf("match=\"%s\"", filterOption),
-					fmt.Sprintf("priority=%v", egressIPReroutePriority),
-					fmt.Sprintf("nexthop=%s", gatewayRouterIP),
-					fmt.Sprintf("external_ids:name=%s", egressIPName),
-					"--",
-					"add",
-					"logical_router",
-					ovnClusterRouter,
-					"policies",
-					"@lr-policy",
-				)
-			}
+			_, stderr, err = util.RunOVNNbctl(
+				"--id=@lr-policy",
+				"create",
+				"logical_router_policy",
+				"action=reroute",
+				fmt.Sprintf("match=\"%s\"", filterOption),
+				fmt.Sprintf("priority=%v", egressIPReroutePriority),
```

Comment:
egress firewall set the reroute priority to less than 1000 (descending), there might be a little chance that egress IP priority is bigger than a egress firewall priority in route policy list. would that be a potential issue although this is unlikely happen statistically?

Reply:
So, I see egress firewall has a start priority of: 10000 https://github.com/ovn-org/ovn-kubernetes/blob/e37c2d9ec3b85c720dc4cc27054def1f4ab0b2ee/go-controller/pkg/ovn/gateway.go#L32 where for each rule in a egress firewall the priority is decreased by one, https://github.com/ovn-org/ovn-kubernetes/blob/e37c2d9ec3b85c720dc4cc27054def1f4ab0b2ee/go-controller/pkg/ovn/egressfirewall.go#L190 I see it has highly improbable that it will ever reach a point where for one EgressFirewall object there are so many rules that the last one gets a lower priority than 100.

Reply:
you're right, the start priority was changed few days ago and I didn't notice the change when commenting.
plus it now has a `minimumReservedEgressFirewallPriority` set and rejects any policy that lower than it.

Reply:
I didn't see that, even better :)